ec2/spark_ec2.py

```diff
@@ -872,9 +890,16 @@ def deploy_files(conn, root_dir, opts, master_nodes, slave_nodes, modules):
     if "." in opts.spark_version:
         # Pre-built Spark deploy
         spark_v = get_validate_spark_version(opts.spark_version, opts.spark_git_repo)
+        tachyon_v = get_tachyon_version(spark_v)
     else:
         # Spark-only custom deploy
         spark_v = "%s|%s" % (opts.spark_git_repo, opts.spark_version)
+        if opts.tachyon_version is None:
```

Comment:
I'll let @shivaram chime in here, but this is a new restriction. Now a user cannot deploy Spark at a specific git hash anymore without also specifying a Tachyon hash.

I'm not sure that makes sense from the outset, but it also breaks users who have been using spark-ec2 just to deploy Spark.